main/src/ca/uwaterloo/flix/util/collection/Chain.scala

```diff
@@ -47,6 +47,25 @@ sealed trait Chain[+A] extends Iterable[A] {
     * The empty chain.
     */
   override val empty: Chain[A] = Chain.Empty
+
+  /**
+    * Returns the amount of elements in the chain.
+    */
+  def length: Int = this match {
+    case Chain.Empty => 0
+    case Chain.Link(l, r) => l.length + r.length
+    case Chain.Many(cs) => cs.map(_.length).sum
+    case Chain.Proxy(xs) => xs.length
+  }
+
+  override def equals(obj: Any): Boolean = obj match {
```

Comment:
I think this risks being too slow. Instead do toList == toList

Also, add a hashcode

Reply:
Because I think calling ViewLeft repeatedly has cost O(n ** 2)

Reply:
I would add a toList (written in CPS) if there is not already one, and then we can use that.

Reply:
Alright roger that. We could also just concatenate all of the inner `Seq`s and compare those?

Reply:
The current problem is that `Iterable` is **not reflexive**.

Reply:
If converted to a list that problem goes away.